api/share/tools.py

```diff
@@ -0,0 +1,28 @@
+from typing import Dict, List, Union
+from http import HTTPStatus
+
+
+def build_correct_http_response(data: Union[Dict, List],
+                                current_url: str,
+                                status: HTTPStatus,
+                                errors: List = None
+                                ) -> Dict:
+    http_response = {
+        "data": data,
+        "_metadata": {
+            "statusCode": status.value,
+            "messages": errors if errors else {"status": status.phrase}
+        },
+        "_links": {
+            "self": current_url
+        }
+    }
+
+    if isinstance(data, dict):
+        http_response["_links"]["collection"] = "/".join(current_url.split("/")[:-2])
```

Comment:
a CSI comment would be nice here. I don't think I understand the intention, and where [:-2] comes from

Reply:
Ok